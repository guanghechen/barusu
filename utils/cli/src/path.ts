import path from 'path'


/**
 * Calc relative path to workspace
 *
 * @param workspace
 * @param targetDir
 */
export function relativeOfWorkspace(
  workspace: string,
  targetDir?: null,
): undefined
export function relativeOfWorkspace(
  workspace: string,
  targetDir: string,
): string
export function relativeOfWorkspace(
  workspace: string,
  targetDir?: string | null,
): string | undefined
export function relativeOfWorkspace(
  workspace: string,
  targetDir?: string | null,
): string | undefined {
  const absoluteDir = absoluteOfWorkspace(workspace, targetDir)
  if (absoluteDir == null) return undefined
  const relativeDir: string = path.relative(workspace, absoluteDir)
  return path.normalize(relativeDir)
}


/**
 * Calc absolute path of p under the workspace
 *
 * @param workspace
 * @param targetDir
 */
export function absoluteOfWorkspace(
  workspace: string,
  targetDir?: null,
): undefined
export function absoluteOfWorkspace(
  workspace: string,
  targetDir: string,
): string
export function absoluteOfWorkspace(
  workspace: string,
  targetDir?: string | null,
): string | undefined
export function absoluteOfWorkspace(
  workspace: string,
  targetDir?: string | null,
): string | undefined {
  if (targetDir == null) return undefined
  const absoluteDir: string = path.resolve(workspace, targetDir)
  return absoluteDir
}
