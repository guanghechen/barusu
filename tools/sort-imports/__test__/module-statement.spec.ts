import path from 'path'
import { runCaseTree } from '@barusu/util-jest'
import {
  createStaticImportOrExportRegexList,
  execWithMultipleRegex,
  extractStaticModuleStatementItem,
  formatStaticModuleStatementItem,
} from '../src'

const regexList = createStaticImportOrExportRegexList('g')
const exec = (
  content: string,
): { lastIndex: number; groups: Record<string, string> } | null => {
  const result = execWithMultipleRegex(regexList, content)
  if (result == null) return null

  return {
    lastIndex: result.regex.lastIndex,
    groups: result.result.groups!,
  }
}

const caseDir = path.resolve(__dirname, 'cases')
runCaseTree(path.resolve(caseDir, 'module-statement.json'), kase => {
  const result = exec(kase.input as string)
  expect({ input: kase.input as string, result }).toMatchSnapshot(
    'execWithMultipleRegex',
  )

  if (result == null) return

  const statementItem = extractStaticModuleStatementItem(result!.groups!)
  expect(statementItem).toMatchSnapshot('extractStaticModuleStatementItem')

  const quote = "'"
  for (const semicolon of [false, true]) {
    for (const indent of ['  ', '\t']) {
      const snapshotName = `formatStaticModuleStatementItem:semicolon(${semicolon}):indent(${indent})`
      const formattedString: string = formatStaticModuleStatementItem(
        statementItem,
        quote,
        indent,
        semicolon,
        80,
      )
      expect(formattedString).toMatchSnapshot(snapshotName)
    }
  }
})
