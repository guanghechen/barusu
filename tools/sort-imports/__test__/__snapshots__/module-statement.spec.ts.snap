// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module-statement default import +import alias: execWithMultipleRegex 1`] = `
Object {
  "input": "import React, { useEffect, useState as waw } from 'react'",
  "result": Object {
    "groups": Object {
      "defaultExport": "React",
      "exportN": "useEffect, useState as waw",
      "keywordFrom": "from",
      "keywordType": undefined,
      "moduleName": "react",
      "quote": "'",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 57,
  },
}
`;

exports[`module-statement default import +import alias: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": "React",
  "exportN": Array [
    "useEffect",
    "useState as waw",
  ],
  "keywordFrom": "from",
  "keywordType": undefined,
  "moduleName": "react",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement default import +import alias: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"import React, { useEffect, useState as waw } from 'react'"`;

exports[`module-statement default import +import alias: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"import React, { useEffect, useState as waw } from 'react'"`;

exports[`module-statement default import +import alias: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"import React, { useEffect, useState as waw } from 'react';"`;

exports[`module-statement default import +import alias: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"import React, { useEffect, useState as waw } from 'react';"`;

exports[`module-statement default import +import all 1: execWithMultipleRegex 1`] = `
Object {
  "input": "import defaultExport, * as name from \\"module-name\\";",
  "result": Object {
    "groups": Object {
      "defaultExport": "defaultExport, * as name",
      "keywordFrom": "from",
      "moduleName": "module-name",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 51,
  },
}
`;

exports[`module-statement default import +import all 1: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": "defaultExport, * as name",
  "exportN": Array [],
  "keywordFrom": "from",
  "keywordType": undefined,
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement default import +import all 1: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"import defaultExport, * as name from 'module-name'"`;

exports[`module-statement default import +import all 1: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"import defaultExport, * as name from 'module-name'"`;

exports[`module-statement default import +import all 1: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"import defaultExport, * as name from 'module-name';"`;

exports[`module-statement default import +import all 1: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"import defaultExport, * as name from 'module-name';"`;

exports[`module-statement default import +import all 2: execWithMultipleRegex 1`] = `
Object {
  "input": "import defaultExport * as name from \\"module-name\\";",
  "result": null,
}
`;

exports[`module-statement default import +import all 3: execWithMultipleRegex 1`] = `
Object {
  "input": "import * as name, defaultExport from \\"module-name\\";",
  "result": null,
}
`;

exports[`module-statement default import base: execWithMultipleRegex 1`] = `
Object {
  "input": "import defaultExport from \\"module-name\\";",
  "result": Object {
    "groups": Object {
      "defaultExport": "defaultExport",
      "keywordFrom": "from",
      "moduleName": "module-name",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 40,
  },
}
`;

exports[`module-statement default import base: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": "defaultExport",
  "exportN": Array [],
  "keywordFrom": "from",
  "keywordType": undefined,
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement default import base: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"import defaultExport from 'module-name'"`;

exports[`module-statement default import base: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"import defaultExport from 'module-name'"`;

exports[`module-statement default import base: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"import defaultExport from 'module-name';"`;

exports[`module-statement default import base: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"import defaultExport from 'module-name';"`;

exports[`module-statement export type base: execWithMultipleRegex 1`] = `
Object {
  "input": "export type { ClassType } from \\"module-name\\";",
  "result": Object {
    "groups": Object {
      "exportN": "ClassType",
      "keywordFrom": "from",
      "keywordType": "type",
      "moduleName": "module-name",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "export",
    },
    "lastIndex": 45,
  },
}
`;

exports[`module-statement export type base: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": undefined,
  "exportN": Array [
    "ClassType",
  ],
  "keywordFrom": "from",
  "keywordType": "type",
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "export",
}
`;

exports[`module-statement export type base: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"export type { ClassType } from 'module-name'"`;

exports[`module-statement export type base: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"export type { ClassType } from 'module-name'"`;

exports[`module-statement export type base: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"export type { ClassType } from 'module-name';"`;

exports[`module-statement export type base: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"export type { ClassType } from 'module-name';"`;

exports[`module-statement export type export type (space before '{' is optional'): execWithMultipleRegex 1`] = `
Object {
  "input": "export type{ ClassType } from \\"module-name\\";",
  "result": Object {
    "groups": Object {
      "exportN": "ClassType",
      "keywordFrom": "from",
      "keywordType": "type",
      "moduleName": "module-name",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "export",
    },
    "lastIndex": 44,
  },
}
`;

exports[`module-statement export type export type (space before '{' is optional'): extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": undefined,
  "exportN": Array [
    "ClassType",
  ],
  "keywordFrom": "from",
  "keywordType": "type",
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "export",
}
`;

exports[`module-statement export type export type (space before '{' is optional'): formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"export type { ClassType } from 'module-name'"`;

exports[`module-statement export type export type (space before '{' is optional'): formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"export type { ClassType } from 'module-name'"`;

exports[`module-statement export type export type (space before '{' is optional'): formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"export type { ClassType } from 'module-name';"`;

exports[`module-statement export type export type (space before '{' is optional'): formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"export type { ClassType } from 'module-name';"`;

exports[`module-statement import alias base: execWithMultipleRegex 1`] = `
Object {
  "input": "import * as name from \\"module-name\\";",
  "result": Object {
    "groups": Object {
      "defaultExport": "* as name",
      "keywordFrom": "from",
      "moduleName": "module-name",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 36,
  },
}
`;

exports[`module-statement import alias base: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": "* as name",
  "exportN": Array [],
  "keywordFrom": "from",
  "keywordType": undefined,
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement import alias base: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"import * as name from 'module-name'"`;

exports[`module-statement import alias base: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"import * as name from 'module-name'"`;

exports[`module-statement import alias base: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"import * as name from 'module-name';"`;

exports[`module-statement import alias base: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"import * as name from 'module-name';"`;

exports[`module-statement import only: execWithMultipleRegex 1`] = `
Object {
  "input": "import \\"module-name\\";",
  "result": Object {
    "groups": Object {
      "moduleName": "module-name",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 21,
  },
}
`;

exports[`module-statement import only: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": undefined,
  "exportN": Array [],
  "keywordFrom": undefined,
  "keywordType": undefined,
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement import only: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"import 'module-name'"`;

exports[`module-statement import only: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"import 'module-name'"`;

exports[`module-statement import only: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"import 'module-name';"`;

exports[`module-statement import only: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"import 'module-name';"`;

exports[`module-statement import type base: execWithMultipleRegex 1`] = `
Object {
  "input": "import type { ClassType } from \\"module-name\\";",
  "result": Object {
    "groups": Object {
      "exportN": "ClassType",
      "keywordFrom": "from",
      "keywordType": "type",
      "moduleName": "module-name",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 45,
  },
}
`;

exports[`module-statement import type base: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": undefined,
  "exportN": Array [
    "ClassType",
  ],
  "keywordFrom": "from",
  "keywordType": "type",
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement import type base: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"import type { ClassType } from 'module-name'"`;

exports[`module-statement import type base: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"import type { ClassType } from 'module-name'"`;

exports[`module-statement import type base: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"import type { ClassType } from 'module-name';"`;

exports[`module-statement import type base: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"import type { ClassType } from 'module-name';"`;

exports[`module-statement import type import type (space before '{' is optional'): execWithMultipleRegex 1`] = `
Object {
  "input": "import type{ ClassType } from \\"module-name\\";",
  "result": Object {
    "groups": Object {
      "exportN": "ClassType",
      "keywordFrom": "from",
      "keywordType": "type",
      "moduleName": "module-name",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 44,
  },
}
`;

exports[`module-statement import type import type (space before '{' is optional'): extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": undefined,
  "exportN": Array [
    "ClassType",
  ],
  "keywordFrom": "from",
  "keywordType": "type",
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement import type import type (space before '{' is optional'): formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"import type { ClassType } from 'module-name'"`;

exports[`module-statement import type import type (space before '{' is optional'): formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"import type { ClassType } from 'module-name'"`;

exports[`module-statement import type import type (space before '{' is optional'): formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"import type { ClassType } from 'module-name';"`;

exports[`module-statement import type import type (space before '{' is optional'): formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"import type { ClassType } from 'module-name';"`;

exports[`module-statement importN +alias 1: execWithMultipleRegex 1`] = `
Object {
  "input": "import { export as alias } from \\"module-name\\";",
  "result": Object {
    "groups": Object {
      "exportN": "export as alias",
      "keywordFrom": "from",
      "keywordType": undefined,
      "moduleName": "module-name",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 46,
  },
}
`;

exports[`module-statement importN +alias 1: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": undefined,
  "exportN": Array [
    "export as alias",
  ],
  "keywordFrom": "from",
  "keywordType": undefined,
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement importN +alias 1: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"import { export as alias } from 'module-name'"`;

exports[`module-statement importN +alias 1: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"import { export as alias } from 'module-name'"`;

exports[`module-statement importN +alias 1: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"import { export as alias } from 'module-name';"`;

exports[`module-statement importN +alias 1: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"import { export as alias } from 'module-name';"`;

exports[`module-statement importN +alias 2: execWithMultipleRegex 1`] = `
Object {
  "input": "import { export1 , export2 as alias2, export3 } from \\"module-name\\";",
  "result": Object {
    "groups": Object {
      "exportN": "export1 , export2 as alias2, export3",
      "keywordFrom": "from",
      "keywordType": undefined,
      "moduleName": "module-name",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 67,
  },
}
`;

exports[`module-statement importN +alias 2: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": undefined,
  "exportN": Array [
    "export1",
    "export2 as alias2",
    "export3",
  ],
  "keywordFrom": "from",
  "keywordType": undefined,
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement importN +alias 2: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"import { export1, export2 as alias2, export3 } from 'module-name'"`;

exports[`module-statement importN +alias 2: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"import { export1, export2 as alias2, export3 } from 'module-name'"`;

exports[`module-statement importN +alias 2: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"import { export1, export2 as alias2, export3 } from 'module-name';"`;

exports[`module-statement importN +alias 2: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"import { export1, export2 as alias2, export3 } from 'module-name';"`;

exports[`module-statement importN +multiple 1: execWithMultipleRegex 1`] = `
Object {
  "input": "import { export1 , export2 } from \\"module-name\\";",
  "result": Object {
    "groups": Object {
      "exportN": "export1 , export2",
      "keywordFrom": "from",
      "keywordType": undefined,
      "moduleName": "module-name",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 48,
  },
}
`;

exports[`module-statement importN +multiple 1: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": undefined,
  "exportN": Array [
    "export1",
    "export2",
  ],
  "keywordFrom": "from",
  "keywordType": undefined,
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement importN +multiple 1: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"import { export1, export2 } from 'module-name'"`;

exports[`module-statement importN +multiple 1: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"import { export1, export2 } from 'module-name'"`;

exports[`module-statement importN +multiple 1: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"import { export1, export2 } from 'module-name';"`;

exports[`module-statement importN +multiple 1: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"import { export1, export2 } from 'module-name';"`;

exports[`module-statement importN +multiple 2: execWithMultipleRegex 1`] = `
Object {
  "input": "import { foo , bar } from \\"module-name/path/to/specific/un-exported/file\\";",
  "result": Object {
    "groups": Object {
      "exportN": "foo , bar",
      "keywordFrom": "from",
      "keywordType": undefined,
      "moduleName": "module-name/path/to/specific/un-exported/file",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 74,
  },
}
`;

exports[`module-statement importN +multiple 2: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": undefined,
  "exportN": Array [
    "bar",
    "foo",
  ],
  "keywordFrom": "from",
  "keywordType": undefined,
  "moduleName": "module-name/path/to/specific/un-exported/file",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement importN +multiple 2: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"import { bar, foo } from 'module-name/path/to/specific/un-exported/file'"`;

exports[`module-statement importN +multiple 2: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"import { bar, foo } from 'module-name/path/to/specific/un-exported/file'"`;

exports[`module-statement importN +multiple 2: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"import { bar, foo } from 'module-name/path/to/specific/un-exported/file';"`;

exports[`module-statement importN +multiple 2: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"import { bar, foo } from 'module-name/path/to/specific/un-exported/file';"`;

exports[`module-statement importN base: execWithMultipleRegex 1`] = `
Object {
  "input": "import { export } from \\"module-name\\";",
  "result": Object {
    "groups": Object {
      "exportN": "export",
      "keywordFrom": "from",
      "keywordType": undefined,
      "moduleName": "module-name",
      "quote": "\\"",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 37,
  },
}
`;

exports[`module-statement importN base: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": undefined,
  "exportN": Array [
    "export",
  ],
  "keywordFrom": "from",
  "keywordType": undefined,
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement importN base: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `"import { export } from 'module-name'"`;

exports[`module-statement importN base: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `"import { export } from 'module-name'"`;

exports[`module-statement importN base: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `"import { export } from 'module-name';"`;

exports[`module-statement importN base: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `"import { export } from 'module-name';"`;

exports[`module-statement multiple line base: execWithMultipleRegex 1`] = `
Object {
  "input": "import { export1, export2, export3 as export3, export4    ,export5, export_long_long_long_long_name } from 'module-name'",
  "result": Object {
    "groups": Object {
      "exportN": "export1, export2, export3 as export3, export4    ,export5, export_long_long_long_long_name",
      "keywordFrom": "from",
      "keywordType": undefined,
      "moduleName": "module-name",
      "quote": "'",
      "remainOfLine": "",
      "type": "import",
    },
    "lastIndex": 120,
  },
}
`;

exports[`module-statement multiple line base: extractStaticModuleStatementItem 1`] = `
Object {
  "defaultExport": undefined,
  "exportN": Array [
    "export1",
    "export2",
    "export3 as export3",
    "export4",
    "export5",
    "export_long_long_long_long_name",
  ],
  "keywordFrom": "from",
  "keywordType": undefined,
  "moduleName": "module-name",
  "remainOfLine": "",
  "type": "import",
}
`;

exports[`module-statement multiple line base: formatStaticModuleStatementItem:semicolon(false):indent(	) 1`] = `
"import {
	export1,
	export2,
	export3 as export3,
	export4,
	export5,
	export_long_long_long_long_name,
} from 'module-name'"
`;

exports[`module-statement multiple line base: formatStaticModuleStatementItem:semicolon(false):indent(  ) 1`] = `
"import {
  export1,
  export2,
  export3 as export3,
  export4,
  export5,
  export_long_long_long_long_name,
} from 'module-name'"
`;

exports[`module-statement multiple line base: formatStaticModuleStatementItem:semicolon(true):indent(	) 1`] = `
"import {
	export1,
	export2,
	export3 as export3,
	export4,
	export5,
	export_long_long_long_long_name,
} from 'module-name';"
`;

exports[`module-statement multiple line base: formatStaticModuleStatementItem:semicolon(true):indent(  ) 1`] = `
"import {
  export1,
  export2,
  export3 as export3,
  export4,
  export5,
  export_long_long_long_long_name,
} from 'module-name';"
`;
